### DISCUSSION TITLE
Overall Solar Yield Vaillant auroMATIC 620

### DISCUSSION THREAD

**Discussion by Bodo1706** [john30/ebusd#824] [Q&A] [answered]
Hi all,

i am searching since days for the parameter which gives me the overall solar yield in kWh. Its shown on the display of the auroMATIC 620, but i couldn't found it on the EBus interface.

If existend, what is the parameter name for the over all solar yield "Solarertrag"?

Greets Bodo

**Comment by kjumybit** [comment, marked as answer]
Hi,
I add a custom message definition in `ec.solsy.sc.csv`.

```
*r,,,,,,B504,,,,,,,
r,,SolarTotalYield,Solarertrag gesamt,,,,13,,,energy;HEX:1;UIN;HEX:1;onoff;HEX:2,,,Solarertrag in kWh mit Status der ED-Solarpumpensteuerung
```
The first value of type `energy` is a total increasing value in kWh.

Example with `ebusctl`:
```
localhost: r -v SolarTotalYield
sc SolarTotalYield energy=368;=00;=120;=00;onoff=off;=01 03

localhost: f SolarTotalYield 
sc SolarTotalYield = 368;00;120;00;off;01 03

localhost: f -h SolarTotalYield 
sc SolarTotalYield = 31ecb5040113 / 09700100780000000103
```
Maybe it works for you.
Bye

**Reply by Bodo1706** [reply]
Hi kjumybit,

works well. Thanks.

Bodo
